core/corehttp/gateway_handler.go

```diff
@@ -301,21 +303,57 @@ func (i *gatewayHandler) getOrHeadHandler(w http.ResponseWriter, r *http.Request
 }
 
 func (i *gatewayHandler) postHandler(w http.ResponseWriter, r *http.Request) {
-	nd, err := i.newDagFromReader(r.Body)
+	fileAdder, err := coreunix.NewAdder(i.node.Context(), i.node, nil)
```

Comment:
we should use the closenotify code like in the get or head requests to make sure we don't have lingering uncanceled contexts from here.